packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -254,6 +275,141 @@ impl Abigen {
     pub fn get_types(abi: &ProgramABI) -> HashMap<usize, TypeDeclaration> {
         abi.types.iter().map(|t| (t.type_id, t.clone())).collect()
     }
+
+    /// Reads the parsed logged types from the ABI and creates
+    /// ResolvedLogs
+    fn resolve_logs(&self) -> Vec<ResolvedLog> {
+        self.abi
+            .logged_types
+            .as_ref()
+            .into_iter()
+            .flatten()
+            .map(|l| {
+                let resolved_type =
+                    resolve_type(&l.application, &self.types).expect("Failed to resolve log type");
+                let param_type_call = single_param_type_call(&resolved_type);
+                let resolved_type_name = TokenStream::from(resolved_type);
+
+                ResolvedLog {
+                    log_id: l.log_id,
+                    param_type_call,
+                    resolved_type_name,
+                }
+            })
+            .collect()
+    }
+}
+
+pub fn generate_print_logs(resolved_logs: &[ResolvedLog]) -> TokenStream {
+    // if not logs are present, return an empty string vec
+    if resolved_logs.is_empty() {
+        return quote! {
+            pub fn _print_logs(&self, receipts: &[Receipt]) -> Vec<String> {
```

Comment:
```suggestion
            pub fn print_logs(&self, receipts: &[Receipt]) -> Vec<String> {
```

Reply:
I think this passed because we are not using it anywhere. Maybe it would make sense to add it to a test ?